plugin/src/test/java/io/jenkins/plugins/analysis/warnings/plugins/MetricsAggregationPluginITest.java

```diff
@@ -0,0 +1,39 @@
+package io.jenkins.plugins.analysis.warnings.plugins;
+
+import org.junit.Test;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.Java;
+import io.jenkins.plugins.metrics.analysis.steps.MetricsRecorder;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+public class MetricsAggregationPluginITest extends IntegrationTestWithJenkinsPerSuite {
+
+    @Test
+    public void shouldAggregateMetrics() {
+        FreeStyleProject project = createJavaWarningsFreestyleProject("pmd.xml", "cpd.xml", "spotbugsXml.xml");
+
+        MetricsRecorder recorder = new MetricsRecorder();
+        recorder.setFilePattern("**/*.txt");
+
+        project.getPublishersList().add(recorder);
+
+        AnalysisResult analysisResult = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
```

Comment:
Build build = scheduleBuild();

build.getAction(Metrics*Action.class).getResults().getMetrics().hasErrorFromWarnings();

Reply:
Diese Action brauchen Sie dann:

https://github.com/andipabst/metrics-aggregation-plugin/blob/ui-plugins/plugin/src/main/java/io/jenkins/plugins/metrics/view/MetricsViewAction.java

Reply:
Soweit ich das sehe, stehen in der MetricsViewAction keine Methoden zur Überprüfung der Results oder Metrics zur Verfügung. Den Build habe ich folgendermaßen erstellt.

     Run<?, ?> build= buildSuccessfully(project);

Die Frage ist jetzt, wie ich mit der Action fortfahren kann.

    build.getAction(MetricsViewAction.class).?

Reply:
Sie müssen die Methode nutzen:
https://github.com/andipabst/metrics-aggregation-plugin/blob/ui-plugins/plugin/src/main/java/io/jenkins/plugins/metrics/view/MetricsViewAction.java#L62

Reply:
Und dann (wahrscheinlich) https://github.com/andipabst/metrics-aggregation-plugin/blob/e9fb720a061da1c06970522f91cedd98d19e413c/plugin/src/main/java/io/jenkins/plugins/metrics/view/MetricsView.java#L91

Evtl. mal im Debugger das Objekt inspizieren...